models/dalec/R/met2model.DALEC.R

```diff
@@ -130,9 +131,9 @@ met2model.DALEC <- function(in.path, in.prefix, outfolder, start_date, end_date,
     }
     
     ##build day of year
-    doy <- rep(1:365,each=86400/dt)
+    doy <- rep(1:365,each=86400/dt)[1:length(sec)]
     if(year %% 4 == 0){  ## is leap
-      doy <- rep(1:366,each=86400/dt)
+      doy <- rep(1:366,each=86400/dt)[1:length(sec)]
```

Comment:
Same comments as above. Note that every time you write `86400/dt`, R has to re-calculate it, which consumes some CPU. Also, when you hard-code it this way everywhere, if it changes, you have to change it everywhere, and if you miss one, it could introduce bugs. Not that the number of seconds per day is ever gonna change, but it's still good practice.